### DISCUSSION TITLE
Using $this when not in object context

### DISCUSSION THREAD

**Discussion by paramwork** [filamentphp/filament#5519] [Help] [answered]
Whenever I use $this in action, I get the "Using $this when not in object context" error.

My controller function is -

```
 public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('name'),
                Tables\Columns\TextColumn::make('email')
                    ->label('Email')
                    ->searchable()
                    ->action(function (WaitlistSubmission $record): void  {
                         $this->dispatchBrowserEvent('open-post-edit-modal', [
                              'post' => $record->getKey(),
                         ])
                    }),                
                Filter::make('created_at')
                    ->form([
                        Forms\Components\DatePicker::make('created_from'),
                        Forms\Components\DatePicker::make('created_until')->default(now()),
                    ])
            ])
            ->actions([
                Tables\Actions\ViewAction::make(),
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ])
            ->defaultSort('created_at', 'desc');
    }

**Comment by danharrin** [comment, marked as answer]
Replace $this with `$livewire` and use `$livewire` in the arguments of that closure

**Reply by ifteakhor** [reply]
Please update the documentation, [Running actions](https://filamentphp.com/docs/3.x/tables/columns/getting-started#running-actions)

**Reply by danharrin** [reply]
The current docs are correct if you are defining a table on a Livewire component or in a relation manager, it's just if you are defining it in a static resource class where this change matters

**Reply by AlyxsonMarques** [reply]
from where this $livewire come from? I have the same problem, but in my case I'm trying to dispatchBrowserEvent from inside a nested repeater on a Resource

![image](https://github.com/user-attachments/assets/1bdb66ba-d677-4622-b162-2412bd014792)

Undefined variable $livewire

**Reply by danharrin** [reply]
You can inject it aside `$get` if you want
